### ISSUE TITLE
Anonymous class declaration and PSR1.Files.SideEffects.FoundWithSymbols

### ISSUE THREAD

**Issue by gsouf** [squizlabs/PHP_CodeSniffer#973]
Declaring an anonymous class (PHP 7 feature) makes rule `PSR1.Files.SideEffects.FoundWithSymbols` to fail.

### Example:

```php
<?php

$foo = 'bar';

$c =  new class extends Something{

    public function someMethod()
    {
        // ...
    }

};
```

phpcs error:

```
/.../test.php:1:1: warning - A file should declare new symbols 
(classes, functions, constants, etc.) and cause no other side effects, 
or it should execute logic with side effects, but should not do both. 
The first symbol is defined on line 7 and the first side effect is on line 3. 
(PSR1.Files.SideEffects.FoundWithSymbols)
```

Please note that the statement alone will also fail:

```php
<?php

new class extends Something{

    public function someMethod()
    {
        // ...
    }

};
```

**Comment by gsherwood** [comment]
The sniff now skips anon classes so it doesn't pick up their methods. Thanks for the report.

**Comment by gsouf** [comment]
Thank you !  Is there a date for the release with the fix ?

**Comment by gsherwood** [comment]
No date, but it probably wont be for a few weeks. I only just made the first commits for 2.6.1 this morning, so I've still got to go through the full bug list and spend time testing.
